### PR TITLE
[EASY] Add warning logs when missing price estimate for a query/token

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
In order to fully get to the root of the issue https://github.com/cowprotocol/services/issues/2814, I would like to add a warning trace for the cases in which a native price for a token couldn't be estimated.

What I want to achieve with that is to know in more details if the high number of `missing_price` failures we have is because the prices aren't available or because the internal logic does not allow fast cache update (or a combination).

### CODE REVIEW

**Review by sunce86 — APPROVED** [review]
Do we want to keep these logs forever or just temporary?

If temporary maybe add a comment TODO to remove later.

**Comment by Mateo-mro** [comment]
> want to keep these logs forever or just temporary?

I wanted to keep them forever if possible, as we don't have logs for it. If it were to be too spammy I could remove it.

File: crates/shared/src/price_estimation/competition/mod.rs [hunk 1]
```diff
@@ -121,6 +121,14 @@ impl<T: Send + Sync + 'static> CompetitionEstimator<T> {
...
                     "new price estimate"
                 );
+                if let Err(ref e) = result {
+                    tracing::warn!(?query,
+                        ?e,
+                        estimator = name,
+                        requests = requests.len(),
+                        elapsed = ?start.elapsed(),
+                        "failed to get price estimate");
+                }
```
Comment (fleupold) [inline crates/shared/src/price_estimation/competition/mod.rs:131]: Don't we already basically have this log by using this query in Kibana?

https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/r/s/2YsfE

File: crates/shared/src/price_estimation/competition/mod.rs [hunk 2]
```diff
@@ -129,6 +137,10 @@ impl<T: Send + Sync + 'static> CompetitionEstimator<T> {
             }
         }
 
+        if results.is_empty() {
```
Comment (fleupold) [inline crates/shared/src/price_estimation/competition/mod.rs:140]: Can results really be empty given that we also push errors into them?

**Review by fleupold — COMMENTED** [review]
Are you sure these logs are useful?

File: crates/shared/src/price_estimation/competition/mod.rs (hunk 1 above)
Comment (Mateo-mro) [reply crates/shared/src/price_estimation/competition/mod.rs:131]: umm you are right, for whatever reason I thought we don't print `debug` and I didn't want to be too spammy in the logs. That will do! thank you!

File: crates/shared/src/price_estimation/competition/mod.rs (hunk 2 above)
Comment (Mateo-mro) [reply crates/shared/src/price_estimation/competition/mod.rs:140]: You are definitely right :facepalm: thank you!

**Pull request closed without merging**